src/session_manager.cpp

```diff
@@ -79,24 +102,60 @@ void SessionManager::handle(Message* msg)
     if (msg->record_type.isInterim())
     {
       // Update the store with the incremented accounting record number.
-      bool success = _store->set_session_data(msg->call_id,
-                                              msg->role,
-                                              msg->function,
-                                              sess,
-                                              msg->trail);
+      bool success = _local_store->set_session_data(msg->call_id,
+                                                    msg->role,
+                                                    msg->function,
+                                                    sess,
+                                                    new_session,
+                                                    msg->trail);
       if (!success)
       {
         // Someone has written conflicting data since we read this, so start processing this message again
         return this->handle(msg);  // LCOV_EXCL_LINE - no conflicts in UT
       }
+
+      std::vector<SessionStore*>::iterator remote_store = _remote_stores.begin();
+
+      while (remote_store != _remote_stores.end())
+      {
+        new_session = false;
+
+        SessionStore::Session* remote_sess = (*remote_store)->get_session_data(msg->call_id,
+                                                                               msg->role,
+                                                                               msg->function,
+                                                                               msg->trail);
+        if (remote_sess == NULL)
+        {
+          remote_sess = new SessionStore::Session();
+          *remote_sess = *sess;
+          new_session = true;
+        }
+        else
+        {
+          remote_sess->acct_record_number += 1;
+        }
+
+        success = (*remote_store)->set_session_data(msg->call_id,
+                                                    msg->role,
+                                                    msg->function,
+                                                    remote_sess,
+                                                    new_session,
+                                                    msg->trail);
+        delete remote_sess; remote_sess = NULL;
+
+        if (success)
+        {
+          ++remote_store;
+        }
```

Comment:
As discussed, you're going to change the store to return a `Store::Status` and update this code to discriminate between `ERROR` and `DATA_CONTENTION`. 

I think this if test either needs some comments, or you should use a double loop to make it clearer what you're doing (loop over stores, loop while in data contention).